hld/session/manager_test.go

```diff
@@ -0,0 +1,448 @@
+package session
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	claudecode "github.com/humanlayer/humanlayer/claudecode-go"
+	"go.uber.org/mock/gomock"
+)
+
+func TestNewManager(t *testing.T) {
+	manager, err := NewManager()
+	if err != nil {
+		t.Fatalf("Failed to create manager: %v", err)
+	}
+
+	if manager == nil {
+		t.Fatal("Manager should not be nil")
+	}
+
+	if manager.sessions == nil {
+		t.Fatal("Sessions map should be initialized")
+	}
+}
+
+func TestSessionLifecycle(t *testing.T) {
+	// Create a manager with empty sessions map
+	manager := &Manager{
+		sessions: make(map[string]*Session),
+	}
+
+	testCases := []struct {
+		name  string
+		setup func()
+		test  func(t *testing.T)
+	}{
+		{
+			name: "list empty sessions",
+			setup: func() {
+				// No setup needed
+			},
+			test: func(t *testing.T) {
+				sessions := manager.ListSessions()
+				if len(sessions) != 0 {
+					t.Errorf("Expected 0 sessions, got %d", len(sessions))
+				}
+
+				infos := manager.ListSessionInfo()
+				if len(infos) != 0 {
+					t.Errorf("Expected 0 session infos, got %d", len(infos))
+				}
+			},
+		},
+		{
+			name: "add and retrieve session",
+			setup: func() {
```

Comment:
i like the split of setup vs test. v clear

when I think of "table test" i usually think of the table entries as "data" more so than functions, and the core test logic is in the `for _, tc := testCases {`

but this feels readable too